subnets.tf

```diff
@@ -0,0 +1,76 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE PUBLIC, PRIVATE, AND INTRA SUBNETS WITHIN THE VPC
+# ---------------------------------------------------------------------------------------------------------------------
+
+locals {
+
+  # hacky assertion
+  assert_subnet_have_unique_group_class_combination = {
+    for subnet in var.subnets :
+    "${subnet.group}-${subnet.class}" => "class/group combination must be unique in all subnets. See the module's README.md for details."
+  }
+
+  subnet_tags = {
+    public  = var.public_subnet_tags
+    private = var.private_subnet_tags
+    intra   = var.intra_subnet_tags
+  }
+
+  # The following creates a list of subnets by constructing multi level lists and flattening them in the end.
+  subnets_tmp = flatten(
+    [
+      for subnet in var.subnets : [
+        for az, netnums in subnet.netnums_by_az : [
+          for idx, netnum in netnums : {
+            cidr_block        = cidrsubnet(try(subnet.cidr_block, var.cidr_block), try(subnet.newbits, 8), netnum)
+            availability_zone = "${local.region}${az}"
+
+            map_public_ip_on_launch = try(subnet.map_public_ip_on_launch, subnet.class == "public", false)
+
+            group_az = "${try(subnet.group, "default")}-${local.region}${az}"
+
+            group = try(subnet.group, "default")
+            class = try(subnet.class, "private")
+            # routes = try(subnet.routes, [])
```

Comment:
can this be removed ?

Reply:
will be implemented in next steps... as routing should be per subnet grouping and class... e.g. main-public routes vs main-private routes... 

in other VPC modules, people tend to route vpc peerings no matter what in all tables.. so let's improve... need to check how this plays together with network ACLs and peerings..